### ISSUE TITLE
RCS1043 - do not raise for types with source generator attributes

### ISSUE THREAD

**Issue by sliekens** [dotnet/roslynator#1678]
This diagnostic is reported incorrectly for types where the other part is generated by a Roslyn source generator like when using `GeneratedRegexAttribute`.

```csharp
partial class Language // RCS1043: Remove 'partial' modifier from type with a single part
{
    [GeneratedRegex("^[a-z]{2}$", RegexOptions.IgnoreCase | RegexOptions.CultureInvariant)]
    private static partial Regex Alpha2Pattern();
}
```
